### PR TITLE
Add chunk & light packet constructor 2

### PR DESCRIPTION
**Pull request by STMiki**
- [x] Packet creation
- [x] Use it to actualy send chunk (need to send it after (I think) the login play packet)

### CODE REVIEW

**Review by Trompettesib — APPROVED** [review]

File: cubic-server/protocol/common.hpp [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 #include <cstdint>
 
+#define LIGHT_ARRAY_SIZE 2048
```
Comment (huntears) [inline cubic-server/protocol/common.hpp:6]: Use a constexpr, not a define

File: cubic-server/protocol/serialization/add.hpp [hunk 2]
```diff
@@ -21,6 +21,31 @@ namespace protocol
...
+    }
+
+    constexpr void addBlockEntities(std::vector<uint8_t> &out, const std::vector<BlockEntity> &data)
+    {
+        addVarInt(out, data.size());
+        for (auto &i : data)
+            addBlockEntity(out, i);
+    }
+
+    constexpr void addLightArray(std::vector<uint8_t> &out, const std::vector<std::array<uint8_t, 2048>> &data)
```
Comment (huntears) [inline cubic-server/protocol/serialization/add.hpp:40]: Use your constant here instead of putting a magic number

**Review by huntears — CHANGES_REQUESTED** [review]

File: cubic-server/protocol/serialization/add.hpp (hunk 2 above)
Comment (STMiki) [reply cubic-server/protocol/serialization/add.hpp:40]: :facepalm:

**Review by Trompettesib — DISMISSED** [review]

File: cubic-server/world_storage/Block.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+world_storage::Block::Block()
+    : name("minecraft:air")
+{
+}
+
+world_storage::Block::Block(const std::string &name, std::unordered_map<std::string, std::string> properties)
+    : name(name), properties(properties)
+{
+    static auto regName = std::regex("([a-z0-9\\.\\-_]+:)?([a-z0-9\\.\\-_/]+)(?:\\[(?:([a-zA-Z_]+=[a-zA-Z0-9_]+),?)*\\])?");
```
Comment (STMiki) [inline cubic-server/world_storage/Block.cpp:12]: TODO: Do this properly lol

**Review by huntears — APPROVED** [review]

**Review by Trompettesib — APPROVED** [review]
GGWP I hope that you didn't loose to much blood nor brain cells

**Review by huntears — APPROVED** [review]

**Review by Trompettesib — APPROVED** [review]

**Pull request merged**